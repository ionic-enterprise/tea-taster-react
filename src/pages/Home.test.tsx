import { render, screen } from '@testing-library/react';
import Home from './Home';

describe('<Home />', () => {
  it('renders consistently', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the title', () => {
    render(<Home />);
    const titleElements = screen.getAllByText('Blank');
    expect(titleElements.length).toEqual(2);
  });

  it('displays the default text', () => {
    const { baseElement } = render(<Home />);
    expect(baseElement).toHaveTextContent(/Ready to create an app?/);
  });
});
