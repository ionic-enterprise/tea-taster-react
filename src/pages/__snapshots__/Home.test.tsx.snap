// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Home /> > renders consistently 1`] = `
<DocumentFragment>
  <div
    class="ion-page"
  >
    {
      "inheritedAttributes": {},
      "setupFadeHeader": [Function],
    }
    {
      "cBottom": -1,
      "cTop": -1,
      "detail": {
        "currentTime": 0,
        "currentX": 0,
        "currentY": 0,
        "data": undefined,
        "deltaX": 0,
        "deltaY": 0,
        "event": undefined,
        "isScrolling": true,
        "scrollLeft": 0,
        "scrollTop": 0,
        "startTime": 0,
        "startX": 0,
        "startY": 0,
        "type": "scroll",
        "velocityX": 0,
        "velocityY": 0,
      },
      "ionScroll": {
        "emit": [Function],
      },
      "ionScrollEnd": {
        "emit": [Function],
      },
      "ionScrollStart": {
        "emit": [Function],
      },
      "isMainContent": true,
      "isScrolling": false,
      "lastScroll": 0,
      "queued": false,
      "resizeTimeout": null,
      "watchDog": null,
    }
  </div>
</DocumentFragment>
`;
